carma_wm_ctrl/test/WMBroadcasterTest.cpp

```diff
@@ -1142,4 +1154,158 @@ TEST(WMBroadcaster, distToNearestActiveGeofence)
+
+  gf->schedules.push_back(carma_wm_ctrl::GeofenceSchedule(ros::Time(1),  // Schedule between 1 and 8
+                                 ros::Time(8),
+                                 ros::Duration(2),    // Starts at 2
+                                 ros::Duration(1.1),  // Ends at by 3.1
+                                 ros::Duration(0),    // 0 offset for repetition start, so still starts at 2
+                                 ros::Duration(1),    // Duration of 1 and interval of two so active durations are (2-3)
+                                 ros::Duration(2)));
+  // convert to ros msg
+  cav_msgs::TrafficControlMessageV01 msg_v01;
+  std::copy(gf->id_.begin(),  gf->id_.end(), msg_v01.id.id.begin());
+  msg_v01.params.schedule.start = gf->schedules[0].schedule_start_;
+  msg_v01.params.schedule.end = gf->schedules[0].schedule_end_;
+  cav_msgs::DailySchedule daily_schedule;
+  daily_schedule.begin = gf->schedules[0].control_start_;
+  daily_schedule.duration = gf->schedules[0].control_duration_;
+  msg_v01.params.schedule.between.push_back(daily_schedule);
+  msg_v01.params.schedule.repeat.offset =  gf->schedules[0].control_offset_;
+  msg_v01.params.schedule.repeat.span =  gf->schedules[0].control_span_;
+  msg_v01.params.schedule.repeat.period =  gf->schedules[0].control_period_;
+
+  ros::Time::setNow(ros::Time(0));  // Set current time
+
+
+
+  
+
+ // variables needed to test
+  std::atomic<uint32_t> map_update_call_count(0);
+  std::atomic<std::size_t> last_active_gf(0);
+  std::atomic<std::size_t> last_inactive_gf(0);
+  bool activated = false;
+
+//Create input message
+  geometry_msgs::PoseStamped input_msg;
+
+  //Input message coordinates
+  input_msg.pose.position.x = 1.5;
+  input_msg.pose.position.y = 1.5;
+  input_msg.pose.position.z = 0.0;
+
+
+
+ size_t base_map_call_count = 0;
+  WMBroadcaster wmb(
+      [&](const autoware_lanelet2_msgs::MapBin& map_bin) {
+        // Publish map callback
+        lanelet::LaneletMapPtr map(new lanelet::LaneletMap);
+        lanelet::utils::conversion::fromBinMsg(map_bin, map);
+      },
+      [&](const autoware_lanelet2_msgs::MapBin& geofence_bin) {
+        auto data_received = std::make_shared<carma_wm::TrafficControl>(carma_wm::TrafficControl());
+        carma_wm::fromBinMsg(geofence_bin, data_received);
+        map_update_call_count.store(map_update_call_count.load() + 1);
+        // atomic is not working for boost::uuids::uuid, so hash it
+        if (activated)
+          last_active_gf.store(boost::hash<boost::uuids::uuid>()(data_received->id_));
+        if (!activated)
+          last_inactive_gf.store(boost::hash<boost::uuids::uuid>()(data_received->id_));
+      }, [](const cav_msgs::TrafficControlRequest& control_msg_pub_){},
+      [](const cav_msgs::CheckActiveGeofence& active_pub_){},
+      std::make_unique<TestTimerFactory>());
+
+   //Test throw exceptions
+  ASSERT_THROW(wmb.currentLocationCallback(input_msg), lanelet::InvalidObjectStateError);
```

Comment:
Making the function private will nto allow you to use this here, but just change it to the logic function. Maybe also we coould change the name of the TEST(WMBroadcaster, currentLocationCallback) as well